src/context/dishes/api-dao.ts

```diff
@@ -0,0 +1,127 @@
+import { err, ok, Result } from 'types/result';
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import {
+  Dish,
+  DishesDao,
+  DishGrade,
+  DishSubmissionRequest,
+  DishSubmissionResponse,
+} from './types';
+import { DefaultDish, DefaultDishSubmissionResponse } from './constants';
+
+const dishesApiDao = ({
+  getIdTokenFunc,
+  baseApiUrl,
+}: {
+  getIdTokenFunc: () => Promise<string>;
+  baseApiUrl: string;
+}): DishesDao => {
+  const axiosClient = axios.create({
+    baseURL: baseApiUrl,
+    timeout: 6000,
+  });
+
+  const getById = async (id: string): Promise<Result<Error, Dish>> => {
+    const request: AxiosRequestConfig = {
+      headers: {
+        Authorization: `Bearer ${await getIdTokenFunc()}`,
```

Comment:
Endpoints that don't need authorization (typically GET endpoints) can probably skip this header. Doesn't hurt to have though, up to you.